bugsnag-android-core/src/androidTest/java/com/bugsnag/android/DeviceIdStoreTest.kt

```diff
@@ -237,4 +252,26 @@ internal class DeviceIdStoreTest {
         assertEquals(prefsId, storeDeviceId)
         assertEquals(prefDeviceId, storeDeviceId)
     }
+
+    /**
+     * If generateAnonymousId is false, no device ID is returned (even if one is saved)
+     */
+    @Test
+    fun loadWithoutGenerateAnonymousId() {
+        file.writeText("{\"id\": \"${ids[0]}\"}")
+        fileInternal.writeText("{\"id\": \"${ids[1]}\"}")
```

Comment:
these look like a good candidate for raw string literals:

```suggestion
        file.writeText("""{"id": "${ids[0]}"}""")
        fileInternal.writeText("""{"id": "${ids[1]}"}""")
```